sdk/search/Azure.Search.Documents/src/Utilities/SearchExtensions.cs

```diff
@@ -68,12 +69,62 @@ public static string GetSearchServiceName(this Uri endpoint)
         /// Split a collection of strings by commas.
         /// </summary>
         /// <param name="value">The value to split.</param>
+        /// <param name="hasODataFunctions">Whether nested OData functions should be ignored.</param>
         /// <returns>A collection of individual values.</returns>
-        public static IList<string> CommaSplit(string value) =>
-            string.IsNullOrEmpty(value) ?
+        public static IList<string> CommaSplit(string value, bool hasODataFunctions = false)
+        {
+            return string.IsNullOrEmpty(value) ?
                 new List<string>() :
-                // TODO: #10600 - Verify we don't need to worry about escaping
-                new List<string>(value.Split(','));
+                new List<string>(hasODataFunctions ? GetClauses() : value.Split(','));
+
+            // See https://docs.microsoft.com/azure/search/search-query-odata-orderby and
+            // https://docs.microsoft.com/en-us/azure/search/search-query-odata-syntax-reference
+            IEnumerable<string> GetClauses()
```

Comment:
While this works, it seems brittle to me. We've been known to extend the syntax of queries without API version bumps, with the expectations that

1. Queries are input-only, and
1. For the case where they round-trip via paging, clients should treat them as opaque strings.

The good news is that the way we typically extend OData syntax is by adding new functions, and this logic should handle that, but long-term this approach is dangerous and may lead to breakage.